sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1240,6 +1133,196 @@ def _update_blocklist_on_error(
 
         return definitely_no_nodes_launched
 
+
+class FailoverCloudErrorHandlerV2:
+    """Handles errors during provisioning and updates the blocked_resources"""
+
+    @staticmethod
+    def _gcp_handler(blocked_resources: Set['resources_lib.Resources'],
```

Comment:
Not immediately getting why we have _gcp_handler() in both V1 and V2 and when they are used respectively.

Reply:
This is mainly because we are using different path for TPU VM creation which have to rely on the v1 handler to handle the errors. We will remove it in #2898. We can either add a comment here or merge #2898 into this pr and merge them together to master.

Reply:
Got it. If we are going to use this PR as an example for new clouds, it seems better to remove the V1 V2 stuff in this PR. Wdyt?

Reply:
In that case, we probably have to merge #2898 in this PR first. I am actually leaning towards to this option as #2898 significantly simplifies the code.

Reply:
We merged #2898 into this branch now. It should be much simplier.